ibis/bigquery/compiler.py

```diff
@@ -0,0 +1,114 @@
+import ibis.sql.compiler as comp
+import ibis.expr.operations as ops
+from ibis.impala.compiler import ImpalaSelect
+from ibis.impala import compiler as impala_compiler
+
+
+class BigQuerySelectBuilder(comp.SelectBuilder):
+
+    @property
+    def _select_class(self):
+        return BigQuerySelect
+
+
+class BigQueryQueryBuilder(comp.QueryBuilder):
+
+    select_builder = BigQuerySelectBuilder
+
+    def __init__(self, expr, context=None, params=None):
+        super(BigQueryQueryBuilder, self).__init__(
+            expr, context=context, params=params
+        )
+
+    def _make_context(self):
+        return BigQueryContext()
+
+    @property
+    def _union_class(self):
+        # return BigQueryUnion
+        raise NotImplementedError()
+
+
+def build_ast(expr, context=None, params=None):
+    builder = BigQueryQueryBuilder(expr, context=context, params=params)
+    return builder.get_result()
+
+
+def _get_query(expr, context, params=None):
+    ast = build_ast(expr, context, params=params)
+    (query, *rest) = ast.queries
+    assert not rest
+    return query
+
+
+def to_sql(expr, context=None, params=None):
+    query = _get_query(expr, context, params=params)
+    compiled = query.compile()
+    return compiled
+
+
+class BigQueryContext(comp.QueryContext):
+
+    def _to_sql(self, expr, ctx):
+        return to_sql(expr, context=ctx)
+
+
+class BigQuerySelect(ImpalaSelect):
+
+    @property
+    def translator(self):
+        return BigQueryExprTranslator
+
+
+def _extract_field(sql_attr):
+    def extract_field_formatter(translator, expr):
+        op = expr.op()
+        arg = translator.translate(op.args[0])
+        return "extract({0!s} from {1!s})".format(sql_attr, arg)
```

Comment:
The format argument indexes and `!s` are unnecessary.

Reply:
I copied the format from the impala compiler.  Can rewrite if desired.

Reply:
No need, we can leave as is.